src/GovUk.Education.ExploreEducationStatistics.Content.Services/DataSetService.cs

```diff
@@ -118,6 +123,57 @@ public DataSetService(ContentDbContext contentDbContext)
             })
             .ToListAsync();
     }
+
+    public async Task<Either<ActionResult, DataSetDetailsViewModel>> GetDataSet(
+        Guid releaseId,
+        Guid fileId)
+    {
+        if (!await _releaseRepository.IsLatestPublishedReleaseVersion(releaseId))
+        {
+            return new NotFoundResult();
+        }
```

Comment:
Looked into this and have discussed it with Ben. 

There is a `userService.CheckCanViewRelease` auth handler, but when I tried to use it that introduces trouble when mocking this - meaning it wouldn't actually have tested the method as thoroughly, as the `IsLatestPublishedReleaseVersion` wouldn't have actually got run against the test data provided.

If there were an easy way to create an `IUserService` via the integration tests, then using the auth handler would have been (almost) equivalent - and this may be possible in the future if the Content API integration tests are set up similar to how the Admin integration tests are.

My overall analysis of the benefits of using the auth handler is that there is really only one benefit: signalling to anyone reading this code that this check is important.

I've added a `WARN` comment above the `IsLatestPublishedReleaseVersion` check.